File: linked list insertion at tail.py

class Node:
  def __init__(self, data):
    self.data = data
    self.next = None
class LinkedList:
  def __init__(self):
    self.head = None
  def push_back(self, newElement):
    newNode = Node(newElement)
    if(self.head == None):
      self.head = newNode
      return
    else:
      temp = self.head
      while(temp.next != None):
        temp = temp.next
      temp.next = newNode
  def PrintList(self):
    temp = self.head
    if(temp != None):
      print("The list contains:", end=" ")
      while (temp != None):
        print(temp.data, end=" ")
        temp = temp.next
      print()
    else:
      print("The list is empty.")
MyList = LinkedList()
MyList.push_back(40)
MyList.push_back(60)
MyList.push_back(80)
MyList.PrintList()

File: linked list insertion at tail.cpp
#include <iostream>
using namespace std;
struct Node {
    int data;
    Node* next;
}
class LinkedList {
  private:
    Node* head;
  public:
    LinkedList(){
      head = NULL;
    }
    void push_back(int newElement) {
      Node* newNode = new Node();
      newNode->data = newElement;
      newNode->next = NULL; 
      if(head == NULL) {
        head = newNode;
      } else {
        Node* temp = head;
        while(temp->next != NULL)
          temp = temp->next;
        temp->next = newNode;
      }    
    }
    void PrintList() {
      Node* temp = head;
      if(temp != NULL) {
        cout<<"The list contains: ";
        while(temp != NULL) {
          cout<<temp->data<<" ";
          temp = temp->next;
        }
        cout<<endl;
      } else {
        cout<<"The list is empty.\n";
      }
    }    
};
int main() {
  LinkedList MyList;
  MyList.push_back(30);
  MyList.push_back(50);
  MyList.push_back(70);
  MyList.PrintList();
  
  return 0; 
}
